app/commands/v2/publish.rb

```diff
@@ -163,6 +167,40 @@ def set_public_updated_at
         end
       end
 
+      def set_temporary_major_published_at
+        return unless update_type == "major"
+
+        edition.update_attributes!(
+          temporary_major_published_at: default_datetime
+        )
+      end
+
+      def set_temporary_minor_published_at
+        return unless update_type == "minor"
+
+        edition.update_attributes!(
+          temporary_minor_published_at: default_datetime
+        )
+      end
+
+      def set_temporary_last_edited_at
+        if update_type.present?
+          edition.update_attributes!(temporary_last_edited_at: default_datetime)
+          return
+        end
+
+        if previous_item.present?
+          edition.update_attributes!(
+            temporary_last_edited_at: previous_item.temporary_last_edited_at
+          )
+        end
```

Comment:
I wonder if we need to do something here if the `previous_item` is not present, i.e. if it was the first publish? Since the state is being edited, do we need to update that here?